src/redist/targets/GenerateLayout.targets

```diff
@@ -492,6 +492,13 @@
 
   </Target>
 
+  <Target Name="GeneratePrecomputedRarCache" DependsOnTargets="LayoutTemplates;LayoutManifests;LayoutBundledTools">
+    <ItemGroup>
+      <AssembliesToResolve Include="$(RedistLayoutPath)\**\*.dll" Exclude="$(RedistLayoutPath)\**\native\**\*.dll" />
+    </ItemGroup>
+    <ResolveAssemblyReference AssemblyFiles="@(AssembliesToResolve)" Silent="false" AssemblyInformationCacheOutputPath="$(RedistLayoutPath)sdk\$(Version)\SDKPrecomputedAssemblyReferences.cache" SearchPaths="{RawFileName}" />
```

Comment:
there is no running stage 2 anymore?

Reply:
The stage 2 MSBuild engine isn't expected to work on ARM, so we can't use the full stage 2 MSBuild as I originally had it. Using the stage 2 tasks assembly (which is what matters) with the stage 0 engine could work, but it isn't great to be intentionally using different versions of the same product in a single call.

This goes a different route and just updates the stage 0 sdk to the minimum we would need to make the cache. Reading the cache is harder—I don't believe that will work with either the stage 0 or stage 2 sdk at this point. It needs https://github.com/dotnet/msbuild/pull/6395 at minimum and https://github.com/dotnet/msbuild/pull/6393 if you installed to Program Files or another folder that requires special write privileges. Fortunately, reading the cache should only happen with the stage 2 sdk, so we won't need a second PR to update the stage 0 sdk again afterwards.

The last viable route we could have gone is to use the stage 0 SDK to build a project that finds the stage 2 tasks assembly and uses that to call RAR. That's a viable option, but it's more complicated (read: I tried for a few hours and didn't get it to work) and slower, and all it would achieve is not needing to update the stage 0 SDK. I don't think updating the stage 0 SDK is bad, so I prefer the current PR.

Let me know if any part of that was confusing 🙂 Long answer to a short question.